bundles/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/internal/converter/SmartthingsHue100Converter.java

```diff
@@ -0,0 +1,96 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.smartthings.internal.dto.SmartthingsStateData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Converter class for Smartthings capability "Color Control".
+ * The Smartthings "Color Control" capability represents the hue values in the 0-100% range. OH2 uses 0-360 degrees
+ * For this converter only the hue is coming into openHAB and it is a number
+ *
+ * @author Bob Raker - Initial contribution
+ */
+@NonNullByDefault
+public class SmartthingsHue100Converter extends SmartthingsConverter {
+
+    private final Logger logger = LoggerFactory.getLogger(SmartthingsHue100Converter.class);
+
+    public SmartthingsHue100Converter(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public String convertToSmartthings(ChannelUID channelUid, Command command) {
+        String jsonMsg;
+
+        if (command instanceof HSBType) {
+            HSBType hsb = (HSBType) command;
+            double hue = hsb.getHue().doubleValue() / 3.60;
+            String value = String.format("[%.0f, %d, %d ]", hue, hsb.getSaturation().intValue(),
+                    hsb.getBrightness().intValue());
+            jsonMsg = String.format(
+                    "{\"capabilityKey\": \"%s\", \"deviceDisplayName\": \"%s\", \"capabilityAttribute\": \"%s\", \"value\": %s}",
+                    thingTypeId, smartthingsName, channelUid.getId(), value);
+        } else {
+            jsonMsg = defaultConvertToSmartthings(channelUid, command);
+        }
+
+        return jsonMsg;
+    }
+
+    @Override
+    public State convertToOpenHab(@Nullable String acceptedChannelType, SmartthingsStateData dataFromSmartthings) {
+        // Here we have to multiply the value from Smartthings by 3.6 to convert from 0-100 to 0-360
+        String deviceType = dataFromSmartthings.capabilityAttribute;
+        Object deviceValue = dataFromSmartthings.value;
+
+        if (deviceValue == null) {
+            logger.warn("Failed to convert Number {} because Smartthings returned a null value.", deviceType);
+            return UnDefType.UNDEF;
+        }
+
+        if ("Number".contentEquals(acceptedChannelType)) {
+            if (deviceValue instanceof String) {
+                double d = Double.parseDouble((String) deviceValue);
+                d *= 3.6;
+                return new DecimalType(d);
+            } else if (deviceValue instanceof Double) {
+                double d = ((Double) deviceValue).doubleValue();
+                d *= 3.6;
+                return new DecimalType(d);
+            } else if (deviceValue instanceof Long) {
+                double d = ((Long) deviceValue).longValue();
+                d *= 3.6;
+                return new DecimalType(d);
```

Comment:
I think that just checking for `Number` instances would be a safer catchall 
```suggestion
            } else if (deviceValue instanceof Number) {
                double d = ((Number) deviceValue).doubleValue();
                d *= 3.6;
                return new DecimalType(d);
            }
```
You should also make a similar change in the `SmartthingsConverter` class.

Reply:
OK to the change to SmartthingsHue100Converter

In SmartthingsConverter are you recommending changing:
```
            case "Number":
                if (deviceValue instanceof String) {
                    return new DecimalType(Double.parseDouble((String) deviceValue));
                } else if (deviceValue instanceof Double) {
                    return new DecimalType((Double) deviceValue);
                } else if (deviceValue instanceof Long) {
                    return new DecimalType((Long) deviceValue);
                } else {
                    logger.warn("Failed to convert Number {} with a value of {} from class {} to an appropriate type.",
                            deviceType, deviceValue, deviceValue.getClass().getName());
                    return UnDefType.UNDEF;
                }

```
To:
```
            case "Number":
                if (deviceValue instanceof String) {
                    return new DecimalType(Double.parseDouble((String) deviceValue));
                } else if (deviceValue instanceof Number) {
                    return new DecimalType((Number) deviceValue);
                } else if (deviceValue instanceof Long) {
                    return new DecimalType((Long) deviceValue);
                } else {
                    logger.warn("Failed to convert Number {} with a value of {} from class {} to an appropriate type.",
                            deviceType, deviceValue, deviceValue.getClass().getName());
                    return UnDefType.UNDEF;
                }

```
This doesn't compile because there is no constructor for **Number**. Or am I missing something?

Reply:
Since DecimalType doesn't have a Number constructor you would need to use one of the other ones.
```
            case "Number":
                if (deviceValue instanceof String) {
                    return new DecimalType(Double.parseDouble((String) deviceValue));
                } else if (deviceValue instanceof Long) {//we catch Long separately to maintain precision if necessary
                    return new DecimalType((Long) deviceValue);
                } else if (deviceValue instanceof Number) {
                    return new DecimalType(((Number) deviceValue).doubleValue());
                } else {
                    logger.warn("Failed to convert Number {} with a value of {} from class {} to an appropriate type.",
                            deviceType, deviceValue, deviceValue.getClass().getName());
                    return UnDefType.UNDEF;
                }
```
If you find it necessary, then you might want add another catch between the Long and Number catches for handling `BigDecimal` instances.

Reply:
Sorry, I a little lost was tro what you want. I have changed the code to be:

```
        if ("Number".contentEquals(acceptedChannelType)) {
            if (deviceValue instanceof String) {
                double d = Double.parseDouble((String) deviceValue);
                d *= 3.6;
                return new DecimalType(d);
            } else if (deviceValue instanceof Long) {
                double d = ((Long) deviceValue).longValue();
                d *= 3.6;
                return new DecimalType(d);
            } else if (deviceValue instanceof BigDecimal) {
                double d = ((BigDecimal) deviceValue).doubleValue();
                d *= 3.6;
                return new DecimalType(d);
            } else if (deviceValue instanceof Number) {
                double d = ((Number) deviceValue).doubleValue();
                d *= 3.6;
                return new DecimalType(d);
            } else {
                logger.warn("Failed to convert Number {} with a value of {} from class {} to an appropriate type.",
                        deviceType, deviceValue, deviceValue.getClass().getName());
                return UnDefType.UNDEF;
            }
        } else {

```

If this isn't right please let me know and I'll change it again.